src/kanata/linux.rs

```diff
@@ -202,3 +222,67 @@ fn handle_scroll(
         _ => unreachable!("expect to be handling a wheel event"),
     }
 }
+
+fn start_event_preprocessor(
+    preprocess_rx: Receiver<KeyEvent>,
+    process_tx: Sender<KeyEvent>,
+    debounce_algorithm: Arc<Mutex<Box<dyn Debounce>>>,
+) {
+    std::thread::spawn(move || {
+        {
+            let algorithm = debounce_algorithm.lock();
+            log::info!(
+                "Starting event preprocessor with debounce algorithm: {}, duration: {} ms",
+                algorithm.name(),
+                algorithm.debounce_time()
+            );
+        }
+        let mut last_tick = Instant::now();
+        let mut ms_remainder_in_ns = 0;
+        let mut has_pending_deadlines = false;
+
+        loop {
+            let mut algorithm = debounce_algorithm.lock();
+
+            if has_pending_deadlines {
+                // Tick every 1ms until no pending deadlines
+                let now = Instant::now();
+                let tick_result = count_ms_elapsed(last_tick, now, ms_remainder_in_ns);
+
+                if tick_result.ms_elapsed >= 1 {
+                    // Call tick if at least 1ms has passed
+                    has_pending_deadlines = algorithm.tick(&process_tx, now);
+                    // Update last_tick and remainder based on the helper function's result
+                    last_tick = Instant::now();
+                    ms_remainder_in_ns = 0;
```

Comment:
- The value assigned to `ms_remainder_in_ns` should be from `tick_result`
- The assignment should not be within the conditional; it should always be assigned from the result